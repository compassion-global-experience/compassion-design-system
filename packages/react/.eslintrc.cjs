module.exports = {
  extends: [
    'standard-jsx',
    'standard-react',
    'plugin:jsx-a11y/recommended',
    'plugin:react/recommended',
    'plugin:react/jsx-runtime',
    'plugin:react-hooks/recommended',
    'plugin:storybook/recommended',
  ],
  env: {
    browser: true,
    es6: true,
    jest: true,
    node: false,
  },
  rules: {
    'jsx-quotes': ['error', 'prefer-double'],
  },
};
