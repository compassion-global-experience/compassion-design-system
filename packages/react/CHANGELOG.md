# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.1.0 (2023-04-08)

### Features

- **react:** css usage and documentation ([11e317c](https://github.com/compassion-global-experience/compassion-design-system/commit/11e317cca4ecdaa385e86d3a5571d7dca554685d))
- **react:** re-export css package ([67549a5](https://github.com/compassion-global-experience/compassion-design-system/commit/67549a58de3de6e02b58c5bdb6f4605001aae783))
- **react:** switch between themes in storybook ([79119ba](https://github.com/compassion-global-experience/compassion-design-system/commit/79119ba48ed3410eaebe9a91ccd076b05c2580c1))
