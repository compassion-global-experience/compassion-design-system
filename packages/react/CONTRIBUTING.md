# Contributing to the `react-web` Package

The `react` package is structured with a component-per-directory convention
inside the `src` directory. Each component directory generally contains 3 items:

- A **TSX file** that describes the component‚Äôs structure and behavior
- A **.stories.ts file** that illustrates most of the ways the component can
  be configured
- An **index.ts** file that exports the TSX component‚Äôs contents

## Workflow

The package uses [Storybook](https://storybook.js.org/docs/react/get-started/introduction)
to facilitate component development and documentation. Storybook uses the files
mentioned above to render components in their various states, and serves as
documentation of those components by surfacing code comments, `propTypes`, and
`defaultProps` values.

To launch Storybook, make sure you‚Äôre in the [/packages/react](/packages/react)
directory and run the command `npm run storybook`. After the project is built,
Storybook will launch in your default browser. New components or modifications
will be displayed in real time! üéâ

The configurations outlined in each component‚Äôs .stories.ts file can be seen in
Storybook‚Äôs **Docs** tab.

## Commit Messages & Versioning

Every package in the compassion-gds monorepo adheres to
[semantic versioning](https://semver.org/). In order to simplify the versioning
and release process, please be sure your commit messages adhere to the
[‚ÄúConventional Commits‚Äù pattern](https://www.conventionalcommits.org/en/v1.0.0/#summary).
You may wish to install the [commitizen/cz-cli](https://github.com/commitizen/cz-cli)
to make this near-bulletproof.

## Making a Pull Request

Once your changes are ready, open a pull request against the `development`
branch of the project, and it will be merged in as quickly as possible!
