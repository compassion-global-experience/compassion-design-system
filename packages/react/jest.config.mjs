/*
 * For a detailed explanation regarding each configuration property, visit:
 * https://jestjs.io/docs/configuration
 */

export default {
  // A preset that is used as a base for Jest's configuration
  preset: 'ts-jest/presets/default-esm',

  // Automatically clear mock calls, instances, contexts and results before every test
  clearMocks: true,

  // Indicates whether the coverage information should be collected while executing the test
  collectCoverage: true,

  // The directory where Jest should output its coverage files
  coverageDirectory: "coverage",

  // Indicates which provider should be used to instrument code for coverage
  coverageProvider: "v8",

  // The test environment that will be used for testing
  testEnvironment: "jsdom",

  // A map from regular expressions to module names or to arrays of module names that allow to stub out resources with a single module
  moduleNameMapper: {
    ".(css|less|scss)$": "identity-obj-proxy",
  },

  // Whether to use watchman for file crawling
  watchman: true,
};
