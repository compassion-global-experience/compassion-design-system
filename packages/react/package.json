{
  "name": "@compassion-gds/react",
  "version": "0.2.0-alpha.2",
  "description": "",
  "type": "module",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "files": [
    "dist"
  ],
  "types": "dist/index.d.ts",
  "exports": {
    ".": {
      "browser": {
        "default": "./dist/esm/index.js"
      },
      "import": "./dist/esm/index.js",
      "require": "./dist/cjs/index.js"
    },
    "./css/base.css": "./dist/css/reset.css",
    "./css/light.css": "./dist/css/cds-light.css",
    "./css/dark.css": "./dist/css/cds-dark.css",
    "./css/*": "./dist/css/*"
  },
  "scripts": {
    "dev": "rollup -c -w",
    "copy-css": "bash ./scripts/copy-css.sh",
    "prepare": "rollup -c && npm run copy-css",
    "test": "jest",
    "storybook": "storybook dev -p 6007",
    "build:storybook": "storybook build --quiet",
    "percy": "cross-env PERCY_TOKEN=$PERCY_TOKEN_REACT percy storybook ./storybook-static",
    "percy-snapshots": "npm run build:storybook && npm run percy"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/compassion-global-experience/compassion-design-system.git",
    "directory": "packages/react"
  },
  "keywords": [
    "compassion",
    "compassion-gds",
    "gds",
    "react"
  ],
  "author": "Compassion XD",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/compassion-global-experience/compassion-design-system/issues"
  },
  "homepage": "https://github.com/compassion-global-experience/compassion-design-system#readme",
  "devDependencies": {
    "@babel/core": "^7.20.12",
    "@babel/preset-env": "^7.20.2",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@compassion-gds/core": "^0.1.0-alpha.0",
    "@rollup/plugin-commonjs": "^24.0.1",
    "@rollup/plugin-node-resolve": "^15.0.1",
    "@rollup/plugin-replace": "^5.0.2",
    "@rollup/plugin-typescript": "^11.1.0",
    "@storybook/addon-actions": "^7.0.2",
    "@storybook/addon-docs": "^7.0.2",
    "@storybook/addon-essentials": "^7.0.2",
    "@storybook/addon-interactions": "^7.0.2",
    "@storybook/addon-links": "^7.0.2",
    "@storybook/preset-react-webpack": "7.0.2",
    "@storybook/react": "7.0.2",
    "@storybook/react-webpack5": "7.0.2",
    "@storybook/testing-library": "^0.0.14-next.1",
    "@testing-library/react": "^13.4.0",
    "@types/jest": "^29.4.0",
    "@types/react": "^18.0.27",
    "babel-loader": "^8.3.0",
    "eslint": "^8.33.0",
    "eslint-config-standard-jsx": "^11.0.0",
    "eslint-config-standard-react": "^13.0.0",
    "eslint-plugin-jsx-a11y": "^6.7.1",
    "eslint-plugin-react": "^7.32.2",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-storybook": "^0.6.10",
    "jest": "^29.4.1",
    "jest-environment-jsdom": "^29.4.1",
    "phosphor-react": "^1.4.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "rollup": "^3.20.2",
    "rollup-plugin-dts": "^5.1.1",
    "rollup-plugin-peer-deps-external": "^2.2.4",
    "rollup-plugin-postcss": "^4.0.2",
    "storybook": "7.0.2",
    "ts-jest": "^29.0.5",
    "webpack": "^5.75.0"
  },
  "peerDependencies": {
    "react": "16.8.x || 17.x || 18.x",
    "react-dom": "16.8.x || 17.x || 18.x"
  }
}
