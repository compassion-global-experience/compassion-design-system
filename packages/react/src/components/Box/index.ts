export { default, BoxProps } from './Box';
