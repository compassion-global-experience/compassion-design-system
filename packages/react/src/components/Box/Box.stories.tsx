import { Meta } from '@storybook/react';
import Box from './Box';

const meta: Meta<typeof Box> = {
  title: 'Components/Box',
  component: Box,
};

export default meta;

const defaultCardContent = (
  <div style={{ padding: '30px' }}>
    <h1>Heading 1</h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
      tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
      veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
      commodo consequat. Duis aute irure dolor in reprehenderit in voluptate
      velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat
      cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id
      est laborum.
    </p>
  </div>
);

export const Default = {
  argTypes: {
    backgroundColor: {
      control: { type: 'select' },
    },
    backgroundTint: {
      control: { type: 'select' },
    },
    addBorder: {
      control: { type: 'boolean' },
    },
    borderRadius: {
      control: { type: 'select' },
    },
    borderColor: {
      control: { type: 'select' },
    },
    isMobile: {
      control: { type: 'boolean' },
    },
    textColor: {
      control: { type: 'select' },
    },
  },

  args: {
    children: defaultCardContent,
  },
};
