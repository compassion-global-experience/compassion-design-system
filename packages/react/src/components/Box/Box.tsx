import { ReactNode } from 'react';
import { getClasses } from '../../utils/classes';
import textStyles from '@compassion-design-system/core/src/components/TextBox/textBox.module.css';
import boxStyles from '@compassion-design-system/core/src/components/Box/box.module.css';

export interface BoxProps {
  children: ReactNode;
  backgroundColor?: 'primary' | 'secondary' | 'accent' | 'neutral';
  backgroundTint?: 'default' | 'subtle' | 'amplified';
  borderRadius?: 'square' | 'rounded' | 'circle';
  borderColor?: 'default' | 'primary';
  isMobile?: boolean;
  textColor?: 'default' | 'primary' | 'white' | 'accent';
  className?: string;
}

const Box = ({
  children,
  isMobile = false,
  textColor = 'default',
  backgroundColor = 'neutral',
  backgroundTint = 'default',
  borderRadius,
  borderColor,
  className,
}: BoxProps) => {
  const boxClasses = [
    textStyles['cds-text-box'],
    textStyles[`cds-text-box--${textColor}`],
    boxStyles['cds-box'],
    boxStyles[`cds-box--${backgroundColor}-${backgroundTint}`],
  ];

  if (borderRadius || borderColor) {
    boxClasses.push(boxStyles['cds-box--border']);
    boxClasses.push(boxStyles[`cds-box--border-${borderRadius || 'square'}`]);
    borderColor !== 'default' &&
      boxClasses.push(boxStyles[`cds-box--border-${borderColor || 'default'}`]);
  }

  if (!isMobile) {
    boxClasses.push(textStyles['cds-text-box--desktop']);
  }

  return (
    <div className={`${boxClasses.join(' ')} ${className || ''}`}>
      <div>{children}</div>
    </div>
  );
};

export default Box;
