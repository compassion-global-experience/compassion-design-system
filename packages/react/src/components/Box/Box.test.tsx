import { render } from '@testing-library/react';
import Box from './Box';

describe('Box', () => {
  it('should render default props correctly', () => {
    const { container } = render(<Box>Default Box</Box>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should add the correct background color classes', () => {
    const { container } = render(
      <Box backgroundColor="primary" backgroundTint="subtle">
        Default Box
      </Box>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should add the correct border classes', () => {
    const { container } = render(
      <Box borderColor="primary" borderRadius="rounded">
        Default Box
      </Box>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should add the correct text color class', () => {
    const { container } = render(<Box textColor="accent">Default Box</Box>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should add the given classname', () => {
    const { container } = render(
      <Box className="testClassName">Default Box</Box>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
