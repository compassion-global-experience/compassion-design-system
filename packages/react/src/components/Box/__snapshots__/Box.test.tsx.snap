// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should add the correct background color classes 1`] = `
<div
  class="cds-text-box cds-text-box--default cds-box cds-box--primary-subtle cds-text-box--desktop "
>
  <div>
    Default Box
  </div>
</div>
`;

exports[`Box should add the correct border classes 1`] = `
<div
  class="cds-text-box cds-text-box--default cds-box cds-box--neutral-default cds-box--border cds-box--border-rounded cds-box--border-primary cds-text-box--desktop "
>
  <div>
    Default Box
  </div>
</div>
`;

exports[`Box should add the correct text color class 1`] = `
<div
  class="cds-text-box cds-text-box--accent cds-box cds-box--neutral-default cds-text-box--desktop "
>
  <div>
    Default Box
  </div>
</div>
`;

exports[`Box should add the given classname 1`] = `
<div
  class="cds-text-box cds-text-box--default cds-box cds-box--neutral-default cds-text-box--desktop testClassName"
>
  <div>
    Default Box
  </div>
</div>
`;

exports[`Box should render default props correctly 1`] = `
<div
  class="cds-text-box cds-text-box--default cds-box cds-box--neutral-default cds-text-box--desktop "
>
  <div>
    Default Box
  </div>
</div>
`;
